hudi-client/hudi-client-common/src/main/java/org/apache/hudi/keygen/KeyGenUtils.java

```diff
@@ -62,7 +62,7 @@
         } else if (kvArray[1].equals(EMPTY_RECORDKEY_PLACEHOLDER)) {
           return "";
         } else {
-          return kvArray[1];
+          return String.join(":", Arrays.copyOfRange(kvArray, 1, kvArray.length));
```

Comment:
In case when one of keys in a complex key has a value which contain `:` character, the earlier implementation would not extract the correct value since it assumes the format is `key:value` with value not containing any `:` character itself. This is a problem in general, and also specifically if timestamp is used for partitioning because it will have `:` character in the time part.

Reply:
Seems like a correct fix.